Automattic/sensei
# Fix notices being displayed on the frontend

Resolves an issue reported in p1698939956077339-slack-C02P7FHLVR9

## Proposed Changes

* This PR is to make sure that frontend notices will be persisted and cleared only on the frontend.
* It fixes some errors with notices being created and persisted on the editor side, while it was supposed to be a frontend check. So the notices were displayed on the frontend from a wrong check.
  * Notice that the issue just happened for users with access to the editor.

The best approach would be to not add that notices through the block rendering, but it could be a very big change, in many parts of the codebase. So I went with the simpler solution (explained above) for now.

## Testing Instructions


1. Create a new course on the editor.
2. Publish it.
3. Navigate to the frontend (it can be the home of the site).
4. Make sure it won't display the notice _"There is no published content in this course yet."_.

## Video

See the error happening:

https://github.com/Automattic/sensei/assets/876340/3fe27a98-102b-41f9-b4bc-ae3834793270

## Pre-Merge Checklist
